sensor/kubernetes/certrefresh/securedcluster_tls_issuer_test.go

```diff
@@ -0,0 +1,474 @@
+				secretsData: map[string]map[string][]byte{sensorSecretName: nil},
+			},
+		},
+		"corrupted data in scanner DB secret": {
+			k8sClientConfig: fakeK8sClientConfig{
+				secretsData: map[string]map[string][]byte{scannerDbSecretName: nil},
+			},
+		},
+		"corrupted data in all secured cluster secrets": {
+			k8sClientConfig: fakeK8sClientConfig{
+				secretsData: map[string]map[string][]byte{
+					sensorSecretName:           nil,
+					collectorSecretName:        nil,
+					admissionControlSecretName: nil,
+					scannerSecretName:          nil,
+					scannerDbSecretName:        nil,
+					scannerV4IndexerSecretName: nil,
+					scannerV4DbSecretName:      nil,
+				},
+			},
+		},
+		"refresh failure and retries": {k8sClientConfig: fakeK8sClientConfig{}, numFailedResponses: 2},
+	}
+
+	for tcName, tc := range testCases {
+		s.Run(tcName, func() {
+			testTimeout := 2 * time.Second
+			ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+			defer cancel()
+			ca, err := mtls.CAForSigning()
+			s.Require().NoError(err)
+
+			secretsCerts := map[string]*mtls.IssuedCert{
+				sensorSecretName:           s.getCertificate(storage.ServiceType_SENSOR_SERVICE),
+				collectorSecretName:        s.getCertificate(storage.ServiceType_COLLECTOR_SERVICE),
+				admissionControlSecretName: s.getCertificate(storage.ServiceType_ADMISSION_CONTROL_SERVICE),
+				scannerSecretName:          s.getCertificate(storage.ServiceType_SCANNER_SERVICE),
+				scannerDbSecretName:        s.getCertificate(storage.ServiceType_SCANNER_DB_SERVICE),
+				scannerV4IndexerSecretName: s.getCertificate(storage.ServiceType_SCANNER_V4_INDEXER_SERVICE),
+				scannerV4DbSecretName:      s.getCertificate(storage.ServiceType_SCANNER_V4_DB_SERVICE),
+			}
+
+			k8sClient := getFakeK8sClient(tc.k8sClientConfig)
+			tlsIssuer := newSecuredClusterTLSIssuer(s.T(), k8sClient, sensorNamespace, sensorPodName)
+			tlsIssuer.certRefreshBackoff = wait.Backoff{
+				Duration: time.Millisecond,
+			}
+
+			s.Require().NoError(tlsIssuer.Start())
+			defer tlsIssuer.Stop(nil)
+			s.Require().NotNil(tlsIssuer.certRefresher)
+			s.Require().False(tlsIssuer.certRefresher.Stopped())
+
+			for i := 0; i < tc.numFailedResponses; i++ {
+				request := s.waitForRequest(ctx, tlsIssuer)
+				response := getSecuredClusterIssueCertsFailureResponse(request.GetRequestId())
+				err = tlsIssuer.ProcessMessage(response)
+				s.Require().NoError(err)
+			}
+
+			request := s.waitForRequest(ctx, tlsIssuer)
+			response := getSecuredClusterIssueCertsSuccessResponse(request.GetRequestId(), ca.CertPEM(), secretsCerts)
+			err = tlsIssuer.ProcessMessage(response)
+			s.Require().NoError(err)
+
+			var secrets *v1.SecretList
+			ok := concurrency.PollWithTimeout(func() bool {
+				secrets, err = k8sClient.CoreV1().Secrets(sensorNamespace).List(context.Background(), metav1.ListOptions{})
+				s.Require().NoError(err)
+				return len(secrets.Items) == 7 && func() bool {
```

Comment:
Personally not a big fan if return statements involving execution of in-line defined anonymous functions.